tests/flow/test_union.py

```diff
@@ -41,3 +64,58 @@ def test_invalid_union(self):
         except redis.exceptions.ResponseError:
             # Expecting an error.
             pass
+
+    # Performing UNION with the same left and right side should
+    # produce the same result as evaluating just one side.
+    def test03_union_deduplication(self):
+        non_union_query = """MATCH (a)-[]->(b) RETURN a.v, b.v ORDER BY a.v, b.v"""
+        non_union_result = redis_graph.query(non_union_query)
+
+        union_query = """MATCH (a)-[]->(b) RETURN a.v, b.v ORDER BY a.v, b.v
+                         UNION
+                         MATCH (a)-[]->(b) RETURN a.v, b.v ORDER BY a.v, b.v"""
+        union_result = redis_graph.query(union_query)
+        self.env.assertEquals(union_result.result_set, non_union_result.result_set)
+
+    # A syntax error should be raised on edge alias reuse in one side of a union.
+    def test04_union_invalid_reused_edge(self):
+        try:
+            query = """MATCH ()-[e]->()-[e]->() RETURN e
```

Comment:
An error should be (and is being omitted) regardless if we're dealing with a UNION query,
```
GRAPH.EXPLAIN g "MATCH ()-[e]->()-[e]->() RETURN e"
(error) Cannot use the same relationship variable 'e' for multiple patterns.
```

Reply:
Correct. This test is to prove that standard AST validations still work when we follow the `_AST_ValidateUnionQuery` code path.